File: Hackerrank/plusMinus.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the plusMinus function below.
function plusMinus(arr) {
    let sz = arr.length;
    let p = 0, n = 0, z = 0;

    for(let i = 0; i <= sz; i++){
        if(arr[i] === 0) z++;
        else if(arr[i] > 0) p++;
        else if(arr[i] < 0) n++;
    }
    console.log(Number.parseFloat(p/sz).toFixed(6));
    console.log(Number.parseFloat(n/sz).toFixed(6));
    console.log(Number.parseFloat(z/sz).toFixed(6));
}

function main() {
    const n = parseInt(readLine(), 10);

    const arr = readLine().split(' ').map(arrTemp => parseInt(arrTemp, 10));

    plusMinus(arr);
}

File: Hackerrank/Plus Minus.java
import java.io.*;
import java.lang.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'plusMinus' function below.
     *
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static void plusMinus(List<Integer> arr) {
    // Write your code here
    //System.out.println(arr);
            int posCount=0;
            int zeroCount=0;
            int negCount=0;
            for(int i=0;i<arr.size();i++){
               // System.out.println(arr.get(i));
                if(arr.get(i)<0){
                    negCount++;
                }else if(arr.get(i)>0){
                    posCount++;
                }else if(arr.get(i)==0){
                    zeroCount++;
                }
            }
           
            
             System.out.printf("%6f%n", (double) posCount / (double) arr.size());
        System.out.printf("%6f%n", (double) negCount / (double) arr.size());
        System.out.printf("%6f%n", (double) zeroCount / (double) arr.size());
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> arr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        Result.plusMinus(arr);

        bufferedReader.close();
    }
}
